cddl/swevidence.cddl

```diff
@@ -0,0 +1,5 @@
+swevidence-type = [+ any]
```

Comment:
If we have an idea of what is acceptable here we should try and constrain the type.  (The type should be extensible)
Something like:
```
swevidence-type = [+ $swevidence-format]

$swevidence-format /= bytes .cbor concise-swid-tag
```
Not sure this would work as-is, but just to give the idea.

Reply:
Hi Thomas,

I mostly like this. I think it works for the document if you add some explanatory text.

But it doesn't work for the CDDL validation part of make for the document because concise-swid-tag is not defined. I have the same problem with the re use of CoSWID's $version-scheme for the HW version claim. Seems there ought to be a convention in CDDL standards docs for handling this sort of thing.

The best I can come up with is more split between the CDDL in the document and the CDDL for validation when running make. 

Any better ideas?

Reply:
> But it doesn't work for the CDDL validation part of make for the document because concise-swid-tag is not defined. I have the same problem with the re use of CoSWID's $version-scheme for the HW version claim. Seems there ought to be a convention in CDDL standards docs for handling this sort of thing.

In CoRIM we [pull CoSWID from the authoritative repo](https://github.com/ietf-rats/ietf-corim-cddl/blob/9d165166f3ae29f2da9e0c261a8289ea58d1d04a/Makefile#L44-L50) and merge it into the final CDDL. This is probably the least worst one can do within the set of constraints of CDDL processing.  I would give this a try first and see if it works.  You want me to test this hypothesis?

Reply:
Pulling CoSWID (or any other) CDDL into the actual published referencing document seems wrong to me. I can see that is being done with CoRIM. It might eventually snowball into the last CDDL standard written having the all the CDDL that came before it copied into it. :-0

However, if the CDDL builds in two forms, one for the document and one for validation, then this works and seems like a good idea. I fixed up EAT so it does that now for CoSWID and SUIT manifests and it is working so far.

I've run into a bit of a name space issue where both EAT and CoSWID define "location". That will eventually be a problem in the CDDL world if we keep doing this.

So thanks for the idea!